cunumeric/module.py

```diff
@@ -3239,6 +3239,9 @@ def put_along_axis(
     Multiple GPUs, Multiple CPUs
 
     """
+
+    if a is not None and a.size == 0:
```

Comment:
@robinw0928 has added some tests when `a`, `values`, and ` indices` are `None`: https://github.com/nv-legate/cunumeric/blob/branch-22.12/tests/integration/test_put_along_axis.py#L239-L253

Reply:
I don't think we need to go out of our way to handle inputs that don't adhere to the function's type signature. In this case the function can expect that `a` will be a (non-`None`) `cunumeric.ndarray`, as the type signature suggests. 

Notably, the presence of the `add_boilerplate` decorator expands this to accept anything `cunumeric-ndarray-like` (e.g. NumPy `ndarray`s, lists and scalars), but that only applies to callers of the function; the function itself can assume that any conversions have already happened, and it's dealing with a `cunumeric.ndarray`.

Correspondingly, I don't think we need to test a function with values that the function doesn't claim to handle.

At the very least, if we do such a test, I don't think it's important to match a specific exception type or message. A code that doesn't expect to handle `None` will just naturally fail at some point when it tries to do something with a `None` object, and the specific mode of failure is arbitrary (whatever happened to be the first thing we tried to do with that object), and can change in the future.

@bryevdv do you agree?

Reply:
I don't think a clear strategy for exception matching has been elucidated. It's a bit tricky, since alot of numpy's exception behavior is probably accidental, but if we want to exactly reproduce exactly, it we will have to be very intentional (to the point of contradicting type sigs as you have noticed).

So how important is exception matching, is the question to answer. Almost certainly there are users somewhere catching specific exceptions from numpy in places, who would see different results if we don't match everything. OTOH maybe it's not that much, and certainly trying to match everything will be very difficult. 

I guess I would say:

* If we don't need to match everything, code like this can go away, along with a handful of the recent QA tests
* It would be  good to document what exactly the goals are (i.e. when do we need to match, and when can we let things slide)

Reply:
Sound good, I will change the code/tests so we don't require to mach error codes.
+1 for the document that will describe the goals